Remove physical-type equality check in reproject_to (#833)
sunpy/ndcube: changelog/833.bugfix.rst

Trivial/Internat Changes
------------------------

- Remove the check for consistent coordinate systems as behaviour controlled in astropy

> -

Reviewer: ```
Removed the check for compatible coordinate systems within ``reproject_to`` as it was redundant.
```

> Trivial/Internat Changes
> ------------------------
> 
> - Remove the check for consistent coordinate systems as behaviour controlled in astropy
> 
> -

Reviewer: ```suggestion
Removed the check for compatible coordinate systems within ``reproject_to`` as it was redundant.
```

Reviewer 2: Or even:
```suggestion
Removed the check for compatible coordinate systems within `~ndcube.NDCube.reproject_to` as it was redundant.
```
